### DISCUSSION TITLE
Can't run ssh from xgterm

### DISCUSSION THREAD

**Discussion by pmassey** [iraf-community/iraf#338] [Q&A] [answered]
One of the things I often did with my old laptop was open up an xgterm window, and then ssh to another machine where my data are stored, and run cl on the other machine, taking advantage of the local xgterm to allow plotting and the like.  However, when I just tried this with the community version of xgterm on my M2 Mac air laptop under Ventura, I get the following error:

ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
ssh_askpass: exec(/usr/X11R6/bin/ssh-askpass): No such file or directory
Permission denied, please try again.

This doesn't happy from a normal terminal window; I can ssh fine.   Similarly, I can ssh from an xterm window.  The only failure is with an xgterm.
 
Perhaps this is related to the 
Warning: no access to tty (Undefined error: 0).
Thus no job control in this shell.

error message one gets when one opens the xgterm?
Thanks for any guidance.  I am operating under tsch, but if I switch to bash I see the same behavior.

**Comment by olebole** [comment, marked as answer]
Can you try without ssh-agent by setting the environment variable `SSH_ASKPASS_REQUIRE` to `never`? Alternatively, you could set the `SSH_ASKPASS` environment variable to the correct executable of ssh-askpass.

**Comment by pmassey** [comment]
Thanks of the suggestion, but I’m afraid that didn’t work: what happens then is that the remote machine asks for my password, but the local machine gives me no opportunity to enter it; it just does this:
Permission denied, please try again.
Permission denied, please try again.
Received disconnect from 10.10.50.143 port 22:2: Too many authentication failures


However, this gave me an idea that solved my problem.   I set up a public/private key using a regular terminal:
On the local machine do a ssh-keygen. Hit a CR for the passphrase
Copy the key to the remote machine: ssh-copy-id -i ~/.ssh/id_rsa.pub username@remotemachine.  When you do this, it will prompt for the password for the remote machine.  Enter it.
Now I find that when I do an ssh it not only doesn’t give me the error message, but I don’t even have to enter the password on the remote machine again.  So, my problem is fixed.  I only try to do this on a very limited number of machines, so I can just set up this key on each of them.

**Reply by olebole** [reply]
You could also try the option `SSH_ASKPASS`.
